src/main/java/org/spongepowered/api/text/Text.java

```diff
@@ -77,6 +80,10 @@
  */
 public abstract class Text implements TextRepresentable {
 
+    static {
+        TypeSerializers.getDefaultSerializers().registerType(TypeToken.of(Text.class), new TextConfigSerializer());
```

Comment:
By making `Text` serializable through Configurate, it should very well be possible that you can make `Text` itself `DataSerializable` since it's very similar with regards to making a `DataBuilder` for a `Text` object. It would certainly allow for using `Text` objects within `DataContainer`s far easier in plugin usage.